### DISCUSSION TITLE
Bad Gateway after installation N AIO behind Traefik

### DISCUSSION THREAD

**Discussion by ghostersk** [nextcloud/all-in-one#1930] [Questions] [answered]
I have Traefik as proxy, and it runs Docker containers on its network.
I can connect to the NextCloud container on port 8080 to set it up. All Dockers are now deployed.
but I still cannot access the Nextcloud on main domain nxc.mydomain.com
I am getting Bad gateway errror. I tried to access it on IP address of the container via SSH port forwarding but no luck either.
Here is my Dockercompose config.

`version: "3.8"
networks:
  traefikauth_net:
    external: true

volumes:
  nextcloud_aio_mastercontainer:
    name: nextcloud_aio_mastercontainer # This line is not allowed to be changed
services:
  nextcloud:
    image: nextcloud/all-in-one:latest
    restart: unless-stopped
    container_name: nextcloud-aio-mastercontainer
    networks:
      - traefikauth_net
    volumes:
      - nextcloud_aio_mastercontainer:/mnt/docker-aio-config
      #- /opt/settings/nextcloud/config:/mnt/docker-aio-config
      - /var/run/docker.sock:/var/run/docker.sock:ro
    ports:
    #  - 80:80
      - 9777:8080
    #  - 8443:8443 
    labels:
      - "traefik.enable=true"                                                                                                                             
      - "traefik.http.middlewares.nextcloud-secure-headers.headers.hostsProxyHeaders=X-Forwarded-Host"                                                                                           
      - "traefik.http.middlewares.nextcloud-secure-headers.headers.hostsProxyHeaders=same-origin"                                                                                           
      - "traefik.http.middlewares.nextcloud-secure-headers.headers.customResponseHeaders.X-Robots-Tag=none"
      - "traefik.http.routers.nextcloud.rule=Host(`nxt.mydomain.com`)"                                                                                                                                   
      - "traefik.http.routers.nextcloud.entrypoints=https"                                                                                                                                                     
      - "traefik.http.routers.nextcloud.tls.certresolver=cloudflare"                                                                                                                                       
      - "traefik.http.routers.nextcloud.service=nextcloud"
      - 'traefik.http.routers.nextcloud.middlewares=authelia@docker'
      - 'traefik.http.routers.nextcloud.middlewares=nextcloud-secure-headers'                                                                                                                               
      - "traefik.http.services.nextcloud.loadBalancer.passHostHeader=true"
      - "traefik.http.services.nextcloud.loadbalancer.server.port=11000"
    environment:
      - APACHE_PORT=11000
      #- APACHE_IP_BINDING=0.0.0.0 # Should be set when running behind a reverse proxy that is running on the same host. See https://github.com/nextcloud/all-in-one/blob/main/reverse-proxy.md
      - SKIP_DOMAIN_VALIDATION=true
      - APACHE_DISABLE_REWRITE_IP=1.
      - NEXTCLOUD_TRUSTED_DOMAINS=< nxt.mydomain.com > < 172.22.0.1 >
      - TRUSTED_PROXIES=< 172.22.0.1 >
      # - COLLABORA_SECCOMP_DISABLED=false # Setting this to true allows to disable Collabora's Seccomp feature. See https://github.com/nextcloud/all-in-one#how-to-disable-collaboras-seccomp-feature
      # - DISABLE_BACKUP_SECTION=false # Setting this to true allows to hide the backup section in the AIO interface.
      - NEXTCLOUD_DATADIR=/opt/settings/nextcloud/datafiles # Allows to set the host directory for Nextcloud's datadir. See https://github.com/nextcloud/all-in-one#how-to-change-the-default-location-of-nextclouds-datadir
      # - NEXTCLOUD_MOUNT=/mnt/ # Allows the Nextcloud container to access the chosen directory on the host. See https://github.com/nextcloud/all-in-one#how-to-allow-the-nextcloud-container-to-access-directories-on-the-host
      # - NEXTCLOUD_UPLOAD_LIMIT=10G # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-upload-limit-for-nextcloud
      # - NEXTCLOUD_MAX_TIME=3600 # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-max-execution-time-for-nextcloud
      # - NEXTCLOUD_MEMORY_LIMIT=512M # Can be adjusted if you need more. See https://github.com/nextcloud/all-in-one#how-to-adjust-the-php-memory-limit-for-nextcloud
      # - NEXTCLOUD_TRUSTED_CACERTS_DIR=/path/to/my/cacerts # CA certificates in this directory will be trusted by the OS of the nexcloud container (Useful e.g. for LDAPS) See See https://github.com/nextcloud/all-in-one#how-to-trust-user-defiend-certification-authorities-ca
      # - NEXTCLOUD_STARTUP_APPS=deck twofactor_totp tasks calendar contacts # Allows to modify the Nextcloud apps that are installed on starting AIO the first time. See https://github.com/nextcloud/all-in-one#how-to-change-the-nextcloud-apps-that-are-installed-on-the-first-startup
      # - NEXTCLOUD_ADDITIONAL_APKS=imagemagick # This allows to add additional packages to the Nextcloud container permanently. Default is imagemagick but can be overwritten by modifying this value. See https://github.com/nextcloud/all-in-one#how-to-add-os-packages-permanently-to-the-nextcloud-container
      # - NEXTCLOUD_ADDITIONAL_PHP_EXTENSIONS=imagick # This allows to add additional php extensions to the Nextcloud container permanently. Default is imagick but can be overwritten by modifying this value. See https://github.com/nextcloud/all-in-one#how-to-add-php-extensions-permanently-to-the-nextcloud-container
      - NEXTCLOUD_ENABLE_DRI_DEVICE=true # This allows to enable the /dev/dri device in the Nextcloud container which is needed for hardware-transcoding. See https://github.com/nextcloud/all-in-one#how-to-enable-hardware-transcoding-for-nextcloud
      # - TALK_PORT=3478 # This allows to adjust the port that the talk container is using.
`

**Comment by Zoey2936** [comment, marked as answer]
AIO does not support treafik configuration via labels at all, please have a look into the reverse proxy documentation
